src/app/pages/security/security.page.ts

```diff
@@ -1,15 +1,127 @@
+    }
+
+    public performChecks(): Promise<any> {
+        return Promise.all<DeviceCheckResult>([
+            this.detectDeviceLock(),
+            this.detectRoot(),
+            this.detectEmulator(),
+            this.detectDebug()]);
+    }
+
+    public performChecksAndPublishMetrics(): Promise<DeviceCheckResult[]> {
+        return this.securityService.checkManyAndPublishMetric(DeviceCheckType.debugModeEnabled,
+            DeviceCheckType.rootEnabled,
+            DeviceCheckType.isEmulator,
+            DeviceCheckType.screenLockEnabled);
+    }
+
+    public addDetection(label: string, isSecure: boolean) {
+        this.totalTests++;
+
+        if (!isSecure) {
+            this.totalDetections++;
+        }
+
+        if (isSecure) { this.totalPassed++; }
+
+        this.detections.push({ label, detected: isSecure });
+        this.trustScore = Number((100 - (((this.totalDetections / this.totalTests) * 100))).toFixed());
+    }
+
+    // Detect if the device is running on an emulator.
+    public detectEmulator(): Promise<any> {
+        return this.securityService.check(DeviceCheckType.isEmulator)
+            .then((isEmulated: DeviceCheckResult) => {
+                const emulatedMsg = isEmulated.passed ? 'Emulator Access Detected' : 'No Emulator Access Detected';
+                this.addDetection(emulatedMsg, !isEmulated.passed);
+            }).catch((err: Error) => console.error(err));
+    }
+
+    // Detect if the device is running Root.
+    public detectRoot(): Promise<any> {
+        return this.securityService.check(DeviceCheckType.rootEnabled)
+            .then((isRooted: DeviceCheckResult) => {
+                const rootedMsg = isRooted.passed ? 'Root Access Detected' : 'No Root Access Detected';
+                this.addDetection(rootedMsg, !isRooted.passed);
+            }).catch((err: Error) => console.error(err));
+    }
+
+    // Detect if the app is running in debug mode.
+    public detectDebug(): Promise<any> {
+        return this.securityService.check(DeviceCheckType.debugModeEnabled)
+            .then((isDebugger: DeviceCheckResult) => {
+                const debuggerMsg = isDebugger.passed ? 'Debugger Detected' : 'No Debugger Detected';
+                this.addDetection(debuggerMsg, !isDebugger.passed);
+            }).catch((err: Error) => console.error(err));
+    }
+
+    // Detect if a system device lock is set.
+    public detectDeviceLock(): Promise<any> {
+        return this.securityService.check(DeviceCheckType.screenLockEnabled)
+            .then((deviceLockEnabled: DeviceCheckResult) => {
+                const deviceLockMsg = deviceLockEnabled.passed ? 'Device Lock Enabled' : 'No Device Lock Enabled';
+                this.addDetection(deviceLockMsg, deviceLockEnabled.passed);
+            });
+    }
+
+    public refreshChecks(): void {
+        this.ionViewWillEnter();
+    }
+
+    public checkDialog(trustScore: number): void {
```

Comment:
[Question] Was requirement to get dialog initially to show this data?

Reply:
it was in the cordova-showcase app. If we don't want it here, I can easily remove it.

Reply:
Cool. No no. I think it will be nicer to see it on the page itself, but that is just my opinion. We can elaborate on that on showcase branch